drift_dev/lib/src/analyzer/sql_queries/type_mapping.dart

```diff
@@ -340,3 +355,27 @@ class TypeMapper {
     }
   }
 }
+
+/// Because nested variables should not be included when extracting all
+/// FoundElements allDescendants.whereType<Variable>() no longer works.
```

Comment:
The "no longer works" comment probably isn't too helpful to someone first reading this in the future :) Maybe we should add a comment to the `if (e is NestedQueryColumn)` explaining that we stop here to avoid reading variables from subqieries.